runtime/lua/vim/diagnostic.lua

```diff
@@ -890,22 +870,25 @@ M.handlers.signs = {
     end
 
     local ns = M.get_namespace(namespace)
-    if not ns.user_data.sign_group then
-      ns.user_data.sign_group = string.format('vim.diagnostic.%s', ns.name)
+    if not ns.user_data.sign_ns then
+      ns.user_data.sign_ns = api.nvim_create_namespace('')
     end
 
-    local sign_group = ns.user_data.sign_group
+    opts.signs.text = opts.signs.text or { 'E', 'W', 'I', 'H' }
```

Comment:
The default should also be smarter than replacing the whole table. For example, if a user only wants to change the sign text for errors but use the default values for the other severities, they might use:

```lua
vim.diagnostic.config({ signs = { text = { [vim.diagnostic.severity.ERROR] = 'X' } } })
```

but this would cause warnings, information, and hints to display as `U`.

Reply:
Something like this (untested):

```lua
local text = opts.signs.text or {}
local default = { [vim.diagnostic.severity.ERROR] = 'E', ... }
for k, v in pairs(default) do
  if not text[k] then
    text[k] = v
  end
end
```

and then use this local `text` table rather than `opts.signs.text` when creating the extmark.